### PR TITLE
Add <ECharts /> component

### PR DESCRIPTION
**Pull request by rwu823**
- Added `fix-unit` utils
- Added webpack `resolve.alias`: `~coms`
- ECharts documentation
 
# ECharts

## Props

```js
props: {
  options: {
    type: Object,
  },
  size: {
    type: [Number, String],
    default: '600x375',
  }
},
```

## Usage

```jsx
import ECharts from '~com/echarts'
render(h) {
  return (
    <div>
      <ECharts size="700x400" options={...echarts options here} />
    </div>
  )
}
```

## Options
All available options:
http://echarts.baidu.com/option.html

## Instance

Access `Chart` inctance from component `$refs`.

```jsx
import ECharts from '~com/echarts'
mounted(){
  this.$refs.chart1.Chart
},
render(h) {
  return (
    <div>
      <ECharts size="700x400" options={...echarts options here} ref="chart1" />
    </div>
  )
}
```
## Events

Each event will be `$emit` form compoent inside, there are all lists of event:
http://echarts.baidu.com/api.html#events

```jsx
import ECharts from '~coms/echarts'

render(h) {
  return (
    <div>
      <ECharts on-eventname={()=>...} size="700x400" options={...echarts options here} />
    </div>
  )
}
```

### CODE REVIEW

File: src/utils/fix-unit.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const fixUnit = (unit) => {
+  if (!unit.endsWith('%')) {
```
Comment (chuanxd) [inline src/utils/fix-unit.js]: 學習惹

File: src/components/echarts.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const { $el, options, _events } = this
+
+    this.Chart = window.echarts.init($el)
+    this.Chart.setOption(options)
+
+    Object.keys(_events).forEach((ev) => {
+      this.Chart.on(ev, (params) => {
+        this.$emit(ev, params)
+      })
+    })
```
Comment (rwu823) [inline src/components/echarts.js:25]: 把 `3rd party library event mapping` 到 vue 上的方法。

File: src/utils/fix-unit.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const fixUnit = (unit) => {
+  if (!/^(string|number)$/.test(typeof unit)) {
```
Comment (chuanxd) [inline src/utils/fix-unit.js:2]: 如果是空字串怎麼辦？

File: src/components/echarts.js (hunk 2 above)
Comment (CApopsicle) [reply src/components/echarts.js:25]: 👍

File: src/utils/fix-unit.js (hunk 3 above)
Comment (rwu823) [reply src/utils/fix-unit.js:2]: 已補上測試
https://github.com/cepave-f2e/owl-light/pull/11/files#diff-50be88e7eb80372bb67f66fb3dc35ced

File: src/utils/fix-unit.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    throw new Error(`${unit} is not a string or number.`)
+  }
+
+  if (typeof unit === 'string' && isNaN(parseFloat(unit))) {
+    throw new Error(`${unit} is not a legal number-string.`)
```
Comment (chuanxd) [inline src/utils/fix-unit.js:7]: 剛想了一下，這邊其實就是處理空字串

File: src/utils/__test__/fix-unit.spec.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    fixUnit('')
+  }).toThrow()
+
+  expect(() => {
+    fixUnit('x123')
+  }).toThrow()
+})
+
+it('If have no unit, it should be return as `px`', () => {
+  expect(fixUnit('87')).toBe('87px')
```
Comment (chuanxd) [inline src/utils/__test__/fix-unit.spec.js:24]: XD

Comment (CApopsicle) [reply src/utils/__test__/fix-unit.spec.js:24]: 😆

**Review by CApopsicle — APPROVED** [review]
LGTM

**Pull request merged**